utils/tx/tx.go

```diff
@@ -6,103 +6,182 @@ import (
-			"the provided Hub RPC is not working, please enter another RPC Endpoint instead (you can be obtained in the following link https://blastapi.io/chains/dymension)",
-		).Show()
+		// Create a new client
+		client, err := cometclient.New(wsURL, "/websocket")
+		if err != nil {
+			return fmt.Errorf("error creating client: %v", err)
+		}
 
-		wsURL = newRPC
-	}
+		// Start the client
+		err = client.Start()
+		if err != nil {
+			return fmt.Errorf("error starting client: %v", err)
+		}
 
-	// Create a new client
-	client, err := cometclient.New(wsURL, "/websocket")
-	if err != nil {
-		return fmt.Errorf("error creating client: %v", err)
-	}
+		// nolint errcheck
+		defer client.Stop()
 
-	// Start the client
-	err = client.Start()
-	if err != nil {
-		return fmt.Errorf("error starting client: %v", err)
-	}
+		// Convert txHash string to bytes
+		txBytes, err := hex.DecodeString(txHash)
+		if err != nil {
+			return fmt.Errorf("error decoding txHash: %v", err)
+		}
 
-	// nolint errcheck
-	defer client.Stop()
+		// Create a query to filter transactions
+		query := fmt.Sprintf("tm.event='Tx' AND tx.hash='%X'", txBytes)
 
-	// Convert txHash string to bytes
-	txBytes, err := hex.DecodeString(txHash)
-	if err != nil {
-		return fmt.Errorf("error decoding txHash: %v", err)
-	}
+		// Subscribe to the query
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+		// nolint errcheck
+		defer cancel()
 
-	// Create a query to filter transactions
-	query := fmt.Sprintf("tm.event='Tx' AND tx.hash='%X'", txBytes)
+		subscription, err := client.Subscribe(ctx, "tx-monitor", query, 100)
+		if err != nil {
+			return fmt.Errorf("error subscribing: %v", err)
+		}
+		// nolint errcheck
+		defer client.Unsubscribe(ctx, "tx-monitor", query)
+
+		fmt.Println("Monitoring transaction:", txHash)
+
+		spinner, _ := pterm.DefaultSpinner.WithText(
+			fmt.Sprintf(
+				"waiting for tx with hash %s to finalize",
+				pterm.FgYellow.Sprint(txHash),
+			),
+		).Start()
+
+		// Listen for events
+		for {
+			select {
+			case event := <-subscription:
+				txEvent, ok := event.Data.(comettypes.EventDataTx)
+				if !ok {
+					fmt.Println("Received non-tx event")
+					continue
+				}
+
+				if txEvent.Result.Code == 0 {
+					spinner.Success("transaction succeeded")
+					pterm.Info.Printf(
+						"Gas wanted: %d, Gas used: %d\n",
+						txEvent.Result.GasWanted,
+						txEvent.Result.GasUsed,
+					)
+					return nil
+				} else {
+					j, _ := json.MarshalIndent(txEvent.Result, "", " ")
+					fmt.Println(string(j))
+
+					return fmt.Errorf("transaction failed with code %d: %v", txEvent.Result.Code, txEvent.Result.Log)
+				}
+			case <-time.After(5 * time.Minute):
+				return fmt.Errorf("timeout waiting for transaction")
+
+			case <-ctx.Done():
+				return fmt.Errorf("context cancelled")
+			}
+		}
+	} else {
+		for {
+			c, _, err := websocket.DefaultDialer.Dial(wsURL, nil)
+			if err != nil {
```

Comment:
this part doesn't check whether the new ws endpoint is responding, does it?

Reply:
it has check and will return error if it can not connect